mistlib/core.py

```diff
@@ -399,6 +399,80 @@ def append_file(input_filename, output_filename):
                 output_file.write(input_content)
     
             return
+    
+    def write_additivefoam_input(self, file):
+        specific_heat_1 = None
+        specific_heat_2 = None
+        # Reference temperature based on a property from self.properties
+        comment_block = """/*---------------------------------------------------------------------------
+     AdditiveFOAM template input file (compatible with 1.0, OpenFOAM 10)
+
+                      Created for simulation with Myna
+  ---------------------------------------------------------------------------*/
+  FoamFile
+{
+    version     2.0;
+    format      ascii;
+    class       dictionary;
+    object      transportProperties;
+}
+
+// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
+
+"""
+
+        reference_temperature = self.properties["solidus_eutectic_temperature"].value
+        p=(self.properties["thermal_conductivity_solid"].value_laurent_poly)
+        labeled_values = {}
+        for idx, pair in enumerate(p, 1):
+            label = f"thermal_cond_S{idx}" 
+            first_value = pair[0]
+            labeled_values[label] = first_value
+       
+        a = []
+        for label, value in labeled_values.items():
+            a.append(value)
+            print(f"{value}")
+        p=(self.properties["thermal_conductivity_liquid"].value_laurent_poly)
+        first_values = [pair[0] for pair in p]
+        labeled_values = {}
+        for idx, pair in enumerate(p, 1):
+            label = f"thermal_cond_S{idx}" 
+            first_value = pair[0]
+            labeled_values[label] = first_value
+        
+        b = []
+        for label, value in labeled_values.items():
+            b.append(value)
+            print(f"{value}")
+
+        p = self.properties["specific_heat_solid"]
+        if (p.value_type == ValueTypes.SCALAR):
+                specific_heat_1 = p.value
+        elif (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                specific_heat_1 = p.evaluate_laurent_polynomial(reference_temperature)
+        else:
+                print("Error: ValueType not supported")
+
+        p = self.properties["specific_heat_liquid"]
+        if (p.value_type == ValueTypes.SCALAR):
+                specific_heat_2 = p.value
+        elif (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                specific_heat_2 = p.evaluate_laurent_polynomial(reference_temperature)
+        else:
+                print("Error: ValueType not supported")
+        
+        with open(file, "w") as f:
+            f.write(comment_block)
+            f.write("solid\n{\n")
+            f.write(f"\tkappa\t ({a[0]} {a[1]} 0.0);\n") 
+            f.write(f"\tCp\t\t ({specific_heat_1} 0.0 0.0);\n")
+            f.write("}\n\n")
+            
+            f.write("liquid\n{\n")
+            f.write(f"\tkappa\t ({b[0]} {b[1]} 0.0);\n") 
+            f.write(f"\tCp\t\t ({specific_heat_2} 0.0 0.0);\n")
```

Comment:
Is there a limitation in additiveFOAM that thermal conductivity has just two terms and the specific heat has just one term? Or is that just what Gerry had in the example? If the additiveFOAM input generally accepts 3 polynomial coefficients, then we should be able to supply as many as they can use.

Reply:
AdditiveFOAM uses the [OpenFOAM Polynomial class](https://cpp.openfoam.org/v10/classFoam_1_1Polynomial.html) here, which can accept N-order polynomials. Practically, I don't anticipate we would use more than a third order polynomial though.

Reply:
@gknapp1, my understanding from @colemanjs is that additiveFOAM is hard-coded to only accept up to third-order polynomials, even if the OpenFOAM class is more general.

@hespuck19, I'm guessing it's on your to-do list and you just haven't gotten to it yet, but I think the main remaining thing to change for this PR is to generalize all of the parameters so that they can be up to third-order polynomials. Although maybe you want `get_property` working before you do that.

Reply:
@stvdwtt you're correct. That's my mistake, I missed the hard-coded polynomial size when I checked the code. 

Though I think it is what you meant, to clarify, AdditiveFOAM is hard-coded to accept 3 polynomial *coefficients* (exponents of 0, 1, and 2), so it would technically only accept a second-order polynomial.

Reply:
@gknapp1, thanks for that clarification, I must have switched three coefficients/third order polynomial in my notes from talking to John.